### PR TITLE
0114 fix crps

### PR DESCRIPTION
**Pull request by davbyr**
1. CRPS stuff now worked with the new system. CRPS parts of unit testing script also work.
3. CRPS method is now faster but uses additional numpy arrays instead of a loop -- so will use more memory. For most applications, memory useage will be small for crps things so not really a problem.
4. CRPS now uses a dataset object to store output and attributes.

*note: this is to merge the 0114 branch into 0104.

EDIT: @jpolton somehow only just noticed your new pull request. Hopefully, none of my unit test changes interfere with yours.

### CODE REVIEW

**Comment by jpolton** [comment]
@davbyr The `unit_test.py` changes mentioned were not committed.

**Comment by davbyr** [comment]
Committed unit testing changes but will now check against ant's before merging.

**Comment by davbyr** [comment]
@jpolton @anwiseNOCL This fixed CRPS branch is ready to be reviewed now. Merging the unit_testing got a little messy in places, but I hope I got it right in the end.

File: coast/CRPS.py [hunk 1]
```diff
@@ -44,111 +49,176 @@ def __init__(self, mod_data, mod_dom, obs_object,
...
-                n_model_pts[ii] = int(mod_subset.shape[0])
+                    # Calculate CRPS and put into output array
+                    crps_list[ii] = mod_cdf.crps_fast(obs_data[ii])
+                    n_model_pts[ii] = int(mod_subset.shape[0])
+                    
+        print("\r Complete.                             \n", end=" ", flush=True)
 
         return crps_list, n_model_pts, contains_land, mod_cdf, obs_cdf
     
+    def subset_indices_by_distance(self, longitude, latitude,
```
Comment (jpolton) [inline coast/CRPS.py:165]: This method appears in NEMO.py and in COAST.py

File: coast/CRPS.py [hunk 2]
```diff
@@ -44,111 +49,176 @@ def __init__(self, mod_data, mod_dom, obs_object,
...
+        # centre. Calls another routine dist_haversine.
+
+        dist = self.calculate_haversine_distance(centre_lon, centre_lat, 
+                                                 longitude, latitude)
+        indices_bool = dist < radius
+        indices = np.where(indices_bool.compute())
+
+        return xr.DataArray(indices[0]), xr.DataArray(indices[1])
+    
+    def calculate_haversine_distance(self, lon1, lat1, lon2, lat2):
```
Comment (jpolton) [inline coast/CRPS.py:189]: This method appear in NEMO.py and in COAST.py

**Pull request merged**